SelfAndPeerEvaluations/partner_eval_quiz_validator.py

```diff
@@ -0,0 +1,457 @@
+        """
+        print(f'Validating "{self.assignment.name}"...')
+        self.__write_quiz_metadata()
+        self.__parse_quiz_questions()
+        with open(
+            f"./quiz_results/quiz_reports/{self.assignment.name}.json", "w"
+        ) as outfile:
+            json.dump(self.quiz_grades, outfile)
+        print(
+            f"Report saved to ./quiz_results/quiz_reports/{self.assignment.name}.json"
+        )
+
+        self.__log_solo_submissions()
+        if create_validation_assignment:
+            self.__create_validation_assignment(validation_assignment_name)
+        print(f"Finish validating {self.assignment.name}")
+
+    def __create_validation_assignment(self, validation_assignment_name: str) -> None:
+        name = (
+            validation_assignment_name
+            if validation_assignment_name != ""
+            else self.assignment.name + " Validator"
+        )
+        print(f'Assigning students "{name}"')
+        validation_assignment = self.course.create_assignment(
+            assignment={
+                "name": name,
+                "description": f"Whether you submitted {self.assignment.name} correctly. If you received a 0, please go back and resubmitt {self.assignment.name} with the commented corrections. This score is totally independent from the grade you will receive from your related project submission.",
+                "assignment_group_id": self.assignment.assignment_group_id,
+                "submission_types": ["none"],
+                "points_possible": 1,
+                "due_at": datetime.datetime.now().isoformat(),
+                "lock_at": datetime.datetime.now().isoformat(),
+                "unlock_at": datetime.datetime.now().isoformat(),
+                "published": True,
+                "omit_from_final_grade": True,
+            }
+        )
+        grade_data = defaultdict(dict)
+        for student_id in self.quiz_grades["info"]["submissions"]:
+            if student_id in self.quiz_errors:
+                grade_data[student_id]["posted_grade"] = 0
+                grade_data[student_id]["text_comment"] = ",\n".join(
+                    [str(error) for error in self.quiz_errors[student_id]]
+                )
+            else:
+                grade_data[student_id]["posted_grade"] = 1
+        validation_assignment.submissions_bulk_update(grade_data=grade_data)
+        if len(self.quiz_errors.keys()) == 0:  # all students submitted perfectly
+            return
+        else:
+            self.assignment.create_override(
+                assignment_override={
+                    "student_ids": list(self.quiz_errors.keys()),
+                    "title": f"{self.assignment.name} make up",
+                    "unlock_at": datetime.datetime.now(),
+                    "due_at": datetime.datetime.now() + datetime.timedelta(days=2),
+                    "lock_at": datetime.datetime.now() + datetime.timedelta(days=2),
```

Comment:
We should probably allow the number of days to allow corrections for to be specified instead of hard coded to 2